File: LR_7-8/3.py
text = input("Текст: ")
result = ""

i = 0
while i < len(text):
    if text[i] == 'x' and text[i+1:i+4] == 'abc':
        result += text[i+1:i+4]
        i += 4
    else:
        result += text[i]
        i += 1

print("Результат:", result)

File: LR/LR_7-8/3.pas
var
  text, result: string;
  i: Integer;
begin
  Write('Текст: ');
  ReadLn(text);
  result := '';

  i := 1;
  while i <= Length(text) do
  begin
    if (text[i] = 'x') and (Copy(text, i + 1, 3) = 'abc') then
    begin
      result := result + 'abc';
      i := i + 4;
    end
    else
    begin
      result := result + text[i];
      i := i + 1;
    end;
  end;

  WriteLn('Результат: ', result);
end.